src/client/src/components/homepage.js

```diff
@@ -0,0 +1,75 @@
+const React = require('react')
+const {Link, Redirect} = require('react-router-dom')
+const fetch = require('node-fetch')
+const logo = require('../logo.svg')
+const responseCodes = require('../statusCodeResponses')
+const LOGIN_URL = '/login'
+
+class HomePage extends React.Component {
+    constructor(props){
+        super(props)
+
+        this.state = {
+            username: '',
+            password: '',
+            successfulLogin: false
+        }
+
+        this.handleChange = this.handleChange.bind(this)
```

Comment:
Yup, due to how React works, this is a common pattern for controlling components. You also need to bind this so that setState works correctly, otherwise your app will likely crash.

See https://reactjs.org/docs/forms.html for more details